prov/gni/src/gnix_vc.c

```diff
@@ -72,6 +73,171 @@ static int __gnix_vc_push_tx_reqs(struct gnix_vc *vc);
  * Helper functions
  ******************************************************************************/
 
+/**
+ * Set key to the given gnix_addr.
+ *
+ * NOTE: If struct gnix_address is ever bit packed or packed by
+ * the compiler this assignment may not set key to the correct
+ * bytes.
+ */
+static inline void __gnix_vc_set_ht_key(void *gnix_addr,
+					gnix_ht_key_t *key)
+{
+	*key = *((gnix_ht_key_t *)gnix_addr);
+}
+
+/**
+ * Look up the vc by fi_addr_t, if it's found just return it,
+ * otherwise allocate a new vc, insert it into the hashtable,
+ * and vector for FI_AV_TABLE AV type, and start connection setup.
+ *
+ * assumptions: ep is non-null;
+ * dest_addr is valid;
+ * vc_ptr is non-null.
+ */
+static int __gnix_vc_get_vc_by_fi_addr(struct gnix_fid_ep *ep, fi_addr_t dest_addr,
+			    struct gnix_vc **vc_ptr)
+{
+	struct gnix_fid_av *av;
+	enum fi_av_type av_type;
+	int ret = FI_SUCCESS;
+	struct gnix_av_addr_entry *av_entry;
+	gnix_ht_key_t key;
+	struct gnix_vc *vc = NULL, *vc_tmp = NULL;
+
+	GNIX_TRACE(FI_LOG_EP_CTRL, "\n");
+
+        GNIX_DEBUG(FI_LOG_EP_CTRL,
+                   "ep->vc_table = %p, ep->vc_table->vector = %p\n",
+                   ep->vc_table, ep->vc_table->vector);
+
+	av = ep->av;
+	if (unlikely(av == NULL)) {
+		GNIX_WARN(FI_LOG_EP_CTRL, "av field NULL for ep %p\n", ep);
+		return -FI_EINVAL;
+	}
+
+	av_type = av->type;
+
+	/*
+	 * if AV for this ep is type FI_AV_TABLE
+	 * do quick lookup, don't bother with taking lock
+	 * as we're only reading.
+	 */
+
+	if (likely(av_type == FI_AV_TABLE)) {
```

Comment:
Yes, my mistake.

> On Apr 20, 2016, at 9:11 AM, Sung-Eun Choi <notifications@github.com> wrote:
> 
> In prov/gni/src/gnix_vc.c:
> 
> > +
> > +	av = ep->av;
> > +	if (unlikely(av == NULL)) {
> > +		GNIX_WARN(FI_LOG_EP_CTRL, "av field NULL for ep %p\n", ep);
> > +		return -FI_EINVAL;
> > +	}
> > +
> > +	av_type = av->type;
> > +
> > +	/*
> > +	 * if AV for this ep is type FI_AV_TABLE
> > +	 * do quick lookup, don't bother with taking lock
> > +	 * as we're only reading.
> > +	 */
> > +
> > +	if (likely(av_type == FI_AV_TABLE)) {
> Is this really likely? Seems like we don't really know.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>